### DISCUSSION TITLE
Meaning of `{.python}`?

### DISCUSSION THREAD

**Discussion by maxdrohde** [quarto-dev/quarto-cli#2887] [Q&A] [answered]
I'm creating a reveal.js presentation for the first time, and I noticed this syntax in the [examples](https://quarto.org/docs/presentations/revealjs/#line-highlighting):

````
```{.python}
1+2
```
````
which works, while the usual syntax without the `.` I've been using in HTML and PDF documents does not:
````
```{python}
1+2
```
````
giving me an error: `ERROR: Invalid jupyter YAML metadata found in file (must be string or object)`

How is `{.python}` different than `{python}` and when should I be using it?

**Comment by mcanouil** [comment, marked as answer]
Both are working properly for me.  
The version with the dot is to output the content without evaluating it.
The following are completely equivalent:

````
```{.python}
1+2
```

```python
1+2
```

```{python}
#| echo: true
#| eval: false
1+2
```
````

**Reply by cscheid** [reply]
````
```{.python}
...
```
````

is the same as

````
```.python
...
```
````

which is the same as

````
```python
...
```
````

This is syntax directly from pandoc. A class declaration by itself has a shorthand without brackets (and a word by itself is shorthanded to a class declaration). The brackets are there to support attributes:

````
```{.python key1=value1 key2=value2}
...
```
````

This is different from

````
```{python}
...
```
````

Which quarto (and knitr) interpret as executable code cells. (Yes, this is moderately confusing.)

**Reply by eitsupi** [reply]
The current documentation is scattered with descriptions of code blocks, requiring users to read the website from cover to cover, I think...
(I remember reading somewhere about code blocks with `{.python}` so I looked on the website but could not find it...)

**Comment by cscheid** [comment]
@maxdrohde Can you share your full .qmd where you're getting a jupyter error? I want to make sure _that_ isn't a bug.

**Reply by maxdrohde** [reply]
@cscheid Thanks for your explanation. I sorted out that issue -- was just an issue with my conda installation of python!

I think what confused me about the {.python} syntax is that it was difficult to find anything about it in the documentation, and the below section in the reveal.js guide made the distinction between executable and not-executable unclear.

https://quarto.org/docs/presentations/revealjs/#line-highlighting

<img width="781" alt="Screen Shot 2022-10-17 at 11 29 23" src="https://user-images.githubusercontent.com/25734232/196232359-eb0f96ac-b354-48be-a698-fc6bc4d4c31c.png">

**Reply by cscheid** [reply]
oof, that sure looks like a typo to me.

**Reply by cscheid** [reply]
Fixed now. Thanks for the report, and sorry for the confusion!
